src/coreclr/jit/simd.h

```diff
@@ -1412,7 +1412,7 @@ void EvaluateWithElementFloating(var_types simdBaseType, TSimd* result, const TS
 
         case TYP_DOUBLE:
         {
-            result->f64[arg1] = static_cast<float>(arg2);
+            result->f64[arg1] = arg2;
```

Comment:
This, and the change in valuenum.cpp are unrelated things I noticed while reading how the SIMD constant folding works.